protocols/kad/src/behaviour.rs

```diff
@@ -446,10 +445,14 @@ where
             provider_record_ttl: config.provider_record_ttl,
             connection_idle_timeout: config.connection_idle_timeout,
             local_addrs: HashSet::new(),
-            caching: config.caching,
         }
     }
 
+    /// Returns the currently configured `replication_factor`.
+    pub(crate) fn replication_factor(&self) -> NonZeroUsize {
```

Comment:
```suggestion
    fn replication_factor(&self) -> NonZeroUsize {
```

Why does this need to be `pub(crate)`? Tests succeed with this diff on my machine.

Reply:
See 24f54bb90.

Reply:
merged, 🙏